tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/transform/C2jModelToGeneratorModelTransformer.java

```diff
@@ -643,12 +906,48 @@ Operation convertOperation(C2jOperation c2jOperation) {
                     }
                 }
             }
+            Map<String, Map<String, String>> operationContextParams = c2jOperation.getOperationContextParams();
+            if (operationContextParams != null )
+            {
+                
+                String path = null;
+                //get path from context param
+                for (Map.Entry<String, Map<String, String>> outerEntry: operationContextParams.entrySet()) {
+                    String outerKey = outerEntry.getKey();
+                    Map<String, String> innerMap = outerEntry.getValue();
+                    String pathValue = innerMap.get("path");
+                    if(pathValue != null)
+                    {
+               
+                        OperationContextCppCodeGenerator ctxt = new OperationContextCppCodeGenerator();
+
+                        String value = JmespathExpression.parse(pathValue)
+                                            .accept(new CppEndpointsJmesPathVisitor(ctxt,
```

Comment:
do we actually need this to be a `CppEndpointsJmesPathVisitor` or can we just traverse the tree like a normal POJO?

Reply:
So, the idea is to use the smithy expression parser which internally traverses the tree after it decomposes the string into different kind of expression. Hence this visitor interface class is needed so that the correct apis are called for appropriate expressions